nixos/modules/services/games/terraria.nix

```diff
@@ -119,7 +119,7 @@ in
       };
 
       dataDir = mkOption {
-        type        = types.str;
+        type        = types.path;
```

Comment:
I don't think this is right, a path willbe copied to the nix store, this should be a string.

Reply:
oh sorry, then I misunderstood the type. I thought this meant that the value needs to start with a "/". I will change it.

Reply:
under `lib/types.nix` I see:

```nix
    path = mkOptionType {
      name = "path";
      check = x: isCoercibleToString x && builtins.substring 0 1 (toString x) == "/";
      merge = mergeEqualOption;
    };
```

where do you get that from that it will be copied to the store ?

Reply:
it's nix's default behavior if people pass `./.` to the options, then nix will add that path to the nix store, then substitute the value with the absolute path to the nix store object.

Reply:
Thank you for your explanation. However, I still get the feeling we are talking
about two different things here.

I know that passing `./.` will result in nix storing this directory in the
store, but, if my understanding of nix serves me correctly, this happens in a
phase before this type checking is actually applied. The typechecker looking at
this declaration would then see a string `"/nix/store/..."`.

What my suggested change in this case would do is prevent people from
accidentally typing `var/lib/terraria`. Granted, this is probably rather rare,
but a more interesting case would be that potential editors of nixos options
know that they can suggest an (existing) path for this option via a filepicker
for example.

There are other options like this already:

`nixos/modules/services/misc/gitweb.nix`:

```nix
projectroot = mkOption {
  default = "/srv/git";
  type = types.path;
  description = ''
    Path to git projects (bare repositories) that should be served by
    gitweb. Must not end with a slash.
  '';
};
```

or

`nixos/modules/services/networking/aria2.nix`:

```nix
downloadDir = mkOption {
  type = types.path;
  default = downloadDir;
  description = ''
      Directory to store downloaded files.
  '';
};
```

Clearly these cannot be copied to the store.

I am not too invested in this, so I will remove it, but one day I would like to
understand this topic a bit better.